src/App.tsx

```diff
@@ -1,24 +1,219 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
+import {
+  getTodos,
+  createTodo,
+  deleteTodo,
+  updateTodo,
+  patchTodoStatus,
+} from './api/todos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { Todo, FilteredBy } from './types/Todo';
+import { TodoNotification } from './components/TodoNotification';
 import { UserWarning } from './UserWarning';
+import { Loader } from './components/Loader';
 
-const USER_ID = 0;
+const USER_ID = 6662;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isTodoLoaded, setIsTodoLoaded] = useState(false);
+  const [filterBy, setFilterBy] = useState<FilteredBy>(FilteredBy.ALL);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isError, setIsError] = useState(false);
+
+  const isActiveTodos = useMemo(
+    () => todos.some(todo => todo.completed),
+    [todos],
+  );
+
+  const countActiveTodos = useMemo(
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
+
+  const getTodosFromServer = async () => {
+    try {
+      setIsTodoLoaded(true);
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+      setIsTodoLoaded(false);
+    } catch (error) {
+      setIsError(true);
+      setIsTodoLoaded(false);
+      setErrorMessage('Data couldn\'t be loaded from the server');
+    }
+  };
+
+  useEffect(() => {
+    getTodosFromServer();
+  }, []);
+
+  const createTodoOnServer = async (query: string) => {
+    const data = {
+      title: query,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setTempTodo({
+      ...data,
+      id: 0,
+    });
+
+    try {
+      setIsTodoLoaded(true);
+      const newTodo = await createTodo(USER_ID, data);
+
+      setTodos([...todos, newTodo]);
+    } catch (error) {
+      setIsError(true);
+      setErrorMessage('Unable to add a todo');
+    } finally {
+      setTempTodo(null);
+      setIsTodoLoaded(false);
+    }
+  };
+
+  const removeTodoOnServer = async (id: number) => {
+    try {
+      setIsTodoLoaded(true);
+      await deleteTodo(id);
+      setTodos(todos.filter(todo => todo.id !== id));
+    } catch (error) {
+      setIsError(true);
+      setErrorMessage('Unable to delete a todo');
+    } finally {
+      setTempTodo(null);
+      setIsTodoLoaded(false);
+    }
+  };
+
+  const updateTodoOnServer = async (todo: Todo, isChange?: boolean) => {
+    const todoId = todo.id;
+    const todoCopy = {
+      ...todo,
+    };
+
+    if (isChange) {
+      todoCopy.completed = !todo.completed;
+    }
+
+    try {
+      setIsTodoLoaded(true);
+      await updateTodo(todoId, todoCopy);
+      await getTodosFromServer();
```

Comment:
This case will gives multiple request. Just update local state without request all todos